src/model/entity/room.go

```diff
@@ -1,10 +1,33 @@
 package entity
 
+type RoomType uint
+
+const (
+	RoomTypeStart RoomType = iota
+	RoomTypeOrdinary
+	RoomTypeFinish
+)
+
 type Room struct {
 	Shape   Box
+	Type    RoomType
 	Foods   []Food
 	Elixirs []Elixir
 	Scrolls []Scroll
 	Weapons []Weapon
 	Enemies []Enemy
+	Portal  *Box
+}
+
+func NewRoom(roomType RoomType, shape Box, portal *Box) Room {
+	return Room{
```

Comment:
В подобных функциях лучше возвращать объект по указателю:


```suggestion
func NewRoom(roomType RoomType, shape Box, portal *Box) *Room {
	return &Room{
```

Reply:
Поправила